File: e2e/playwright-python/providers/step_provider.py
from playwright.sync_api import Page
from interactions.steps.navigation_step import NavigationStep
from interactions.steps.task_table_step import TaskTableStep
from interactions.steps.language_switcher_step import LanguageSwitcherStep
from interactions.steps.accessibility_step import AccessibilityStep


class StepProvider:
    def __init__(self, page: Page, is_mobile: bool):
        self.navigation = NavigationStep(page)
        self.tasks = TaskTableStep(page)
        self.language = LanguageSwitcherStep(page)
        self.accessibility = AccessibilityStep(page)

File: e2e/cucumber-playwright-typescript/providers/StepProvider.ts
import { APIRequestContext, Page } from '@playwright/test';
import { AccessibilityStep } from '@/interactions/steps/AccessibilityStep';
import { CreateTaskStep } from '@/interactions/steps/CreateTaskStep';
import { EditTaskStep } from '@/interactions/steps/EditTaskStep';
import { LanguageSwitcherStep } from '@/interactions/steps/LanguageSwitcherStep';
import { NavigationStep } from '@/interactions/steps/NavigationStep';
import { TaskTableStep } from '@/interactions/steps/TaskTableStep';

export class StepProvider {
  readonly navigation: NavigationStep;
  readonly tasks: TaskTableStep;
  readonly createTask: CreateTaskStep;
  readonly editTask: EditTaskStep;
  readonly language: LanguageSwitcherStep;
  readonly accessibility: AccessibilityStep;

  constructor(page: Page, _isMobile: boolean, _request: APIRequestContext) {
    this.navigation = new NavigationStep(page);
    this.tasks = new TaskTableStep(page);
    this.createTask = new CreateTaskStep(page);
    this.editTask = new EditTaskStep(page);
    this.language = new LanguageSwitcherStep(page);
    this.accessibility = new AccessibilityStep(page);
  }
}
